src/Features/CSharp/Portable/Completion/KeywordRecommenders/InKeywordRecommender.cs

```diff
@@ -15,13 +17,63 @@ public InKeywordRecommender()
+            {
+                SyntaxKind.AbstractKeyword,
+                // SyntaxKind.AsyncKeyword,    // async methods cannot be in
+                SyntaxKind.ExternKeyword,
+                SyntaxKind.InternalKeyword,
+                SyntaxKind.NewKeyword,
+                SyntaxKind.OverrideKeyword,
+                SyntaxKind.PublicKeyword,
+                SyntaxKind.PrivateKeyword,
+                SyntaxKind.ProtectedKeyword,
+                // SyntaxKind.ReadOnlyKeyword, // fields cannot be in
+                SyntaxKind.SealedKeyword,
+                SyntaxKind.StaticKeyword,
+                SyntaxKind.UnsafeKeyword,
+                SyntaxKind.VirtualKeyword,
+                // SyntaxKind.VolatileKeyword, // fields cannot be in
+            };
+
+        /// <summary>
+        /// Same as <see cref="SyntaxKindSet.AllGlobalMemberModifiers"/> with in specific exclusions
+        /// </summary>
+        private static readonly ISet<SyntaxKind> InGlobalMemberModifiers = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
+            {
+                // SyntaxKind.AsyncKeyword,    // async methods cannot be in
+                SyntaxKind.ExternKeyword,
+                SyntaxKind.InternalKeyword,
+                SyntaxKind.NewKeyword,
+                SyntaxKind.OverrideKeyword,
+                SyntaxKind.PublicKeyword,
+                SyntaxKind.PrivateKeyword,
+                // SyntaxKind.ReadOnlyKeyword, // fields cannot be in
+                SyntaxKind.StaticKeyword,
+                SyntaxKind.UnsafeKeyword,
+                // SyntaxKind.VolatileKeyword, // fields cannot be in
+            };
+
         protected override bool IsValidContext(int position, CSharpSyntaxContext context, CancellationToken cancellationToken)
         {
             return
                 IsValidContextInForEachClause(context) ||
                 IsValidContextInFromClause(context, cancellationToken) ||
                 IsValidContextInJoinClause(context, cancellationToken) ||
-                context.TargetToken.IsTypeParameterVarianceContext();
+                context.TargetToken.IsTypeParameterVarianceContext() ||
+                context.SyntaxTree.IsParameterModifierContext(position, context.LeftToken, cancellationToken) ||
+                context.SyntaxTree.IsAnonymousMethodParameterModifierContext(position, context.LeftToken, cancellationToken) ||
+                context.SyntaxTree.IsPossibleLambdaParameterModifierContext(position, context.LeftToken, cancellationToken) ||
+                context.IsDelegateReturnTypeContext ||
+                context.SyntaxTree.IsGlobalMemberDeclarationContext(position, InGlobalMemberModifiers, cancellationToken) ||
+                context.IsMemberDeclarationContext(
+                    validModifiers: InMemberModifiers,
+                    validTypeDeclarations: SyntaxKindSet.ClassInterfaceStructTypeDeclarations,
+                    canBePartial: true,
```

Comment:
Good catch!

---
In reply to: [103393260](https://github.com/dotnet/roslyn/pull/17447#discussion_r103393260) [](ancestors = 103393260)